File: heap/heap_test.go
package heap

import (
	"math/rand"
	"testing"
)

func less(i, j int) bool {
	return i < j
}

func TestHeap(t *testing.T) {
	heap := NewHeap(20, less)
	for range 100 {
		heap.Add(rand.Intn(100))
	}

	sorted := heap.Sorted()
	for i := range 19 {
		if sorted[i] < sorted[i+1] {
			t.Fail()
		}
	}
}

File: Julia/heap_test.jl
using Test

include("heap.jl")

less(i, j) = i < j

function test_heap()
    heap = Vector{Int8}()
    items = rand(Int8, 100)
    for item in items
        heap_push!(heap, item, 20, less)
    end
    length(heap) == 20 || return false
    for i in 2:20
        heap[i÷2] <= heap[i] || return false
    end
    return true
end

@test test_heap()